lib/buffer.js

```diff
@@ -246,18 +247,30 @@ function fromArrayLike(obj) {
   const length = obj.length;
   const b = allocate(length);
   for (var i = 0; i < length; i++)
-    b[i] = obj[i] & 255;
+    b[i] = obj[i];
   return b;
 }
 
 function fromArrayBuffer(obj, byteOffset, length) {
+  if (!isArrayBuffer(obj))
+    throw new TypeError('argument is not an ArrayBuffer');
```

Comment:
That test even passes without this check now, the expected TypeError is thrown when accessing `byteLength` below, although with a different message (which would *technically* make dropping this a semver-major change…).

I assume the `isArrayBuffer` check was only added in the first place to avoid crashes on the C++ side, but that doesn’t seem to be a concern anymore if everything’s done in JS.